samples/sample_common/src/vaapi_allocator.cpp

```diff
@@ -438,47 +438,46 @@ mfxStatus vaapiFrameAllocator::LockFrame(mfxMemId mid, mfxFrameData *ptr)
 
         if (MFX_ERR_NONE == mfx_res)
         {
-            va_res = m_libva->vaMapBuffer(m_dpy, vaapi_mid->m_image.buf, (void **) &pBuffer);
+            va_res = m_libva->vaMapBuffer(m_dpy, vaapi_mid->m_image.buf, (void **)&pBuffer);
             mfx_res = va_to_mfx_status(va_res);
         }
         if (MFX_ERR_NONE == mfx_res)
         {
             switch (vaapi_mid->m_image.format.fourcc)
             {
             case VA_FOURCC_NV12:
-                if (mfx_fourcc == MFX_FOURCC_NV12)
+                if (mfx_fourcc != vaapi_mid->m_image.format.fourcc)
```

Comment:
how about other cases? VA_FOURCC_UYVY, VA_FOURCC_RGB565 and so on

Reply:
For RGB565 and UYVY we have another cases.